### ISSUE TITLE
Proxy NCA And Softmax Scale / Label Smoothing

### ISSUE THREAD

**Issue by lou-k** [KevinMusgrave/powerful-benchmarker#90]
First off, thanks for the repository and the paper! I had some trouble with reproducing some of these in the past, and it was very validating to know that there wasn't actual increases.

I used the repository to reproduce the No Fuss paper (i.e., proxyNCA), but I noticed that you include a `softmax_scale` parameter to scale the logits. I couldn't find any documentation about the motivation behind it, but noticed that without it, the performance is off by about 4%. 

Could you provide any intuition behind the scaling? I know the original paper included label smoothing but that doesn't seem to help here. Thanks!

**Comment by KevinMusgrave** [comment]
Larger values of ```softmax_scale``` result in a peakier distribution:

```python
import torch
import torch.nn.functional as F
x = torch.tensor([[0, 0.5, 1]])

F.softmax(x, dim=1)
# tensor([[0.1863, 0.3072, 0.5065]])

F.softmax(x*2, dim=1)
# tensor([[0.0900, 0.2447, 0.6652]])

F.softmax(x*0.5, dim=1)
# tensor([[0.2543, 0.3265, 0.4192]])
```

Many papers use the term "temperature", which is the inverse of ```softmax_scale```.
Here's one explanation from https://arxiv.org/pdf/1811.12649.pdf:

![image](https://user-images.githubusercontent.com/2314891/127660881-f6f66f1d-0af7-44ad-8952-83cab3614e9a.png)

Other papers that use temperature:
[MoCo](https://arxiv.org/pdf/1911.05722.pdf)
[SimCLR](https://arxiv.org/pdf/2002.05709.pdf)

**Comment by KevinMusgrave** [comment]
This paper talks about temperature scaling with proxy nca: https://arxiv.org/pdf/2004.01113.pdf

**Comment by lou-k** [comment]
Ahhh thanks so much! I clearly was a little behind on the literature.
